qiskit_nature/second_q/algorithms/ground_state_solvers/ground_state_eigensolver.py

```diff
@@ -52,23 +45,17 @@ def __init__(
         self._solver = solver
 
     @property
-    def solver(self) -> Union[MinimumEigensolver, MinimumEigensolverFactory]:
-        """Returns the minimum eigensolver or factory."""
+    def solver(self):
         return self._solver
```

Comment:
I would like to improve the handling of the `solver` property. It is a bit all over the place with the union of the actual solver and its factory right now.